pg_shard.c

```diff
@@ -713,25 +768,123 @@ ClassifyRestrictions(List *queryRestrictList, List **remoteRestrictList,
 
 
 /*
- * RowAndColumnFilterQuery builds a query which contains the filter clauses from
+ * ClassifyHasAggregates decides whether local and distributed queries will include
+ * aggregates or not. remoteQueryHasAggregates and localQueryHasAggregates are output
+ * parameters to receive these boolean values.
+ */
+static void
+ClassifyHasAggregates(Query *query, bool *remoteQueryHasAggregates,
```

Comment:
Nothing is really being classified here. I'd say `DetermineRemoteAndLocalAggregatePresence`. Is that too long? It is much better than classify.